src/QsCompiler/Compiler/Templates/QirDriverCpp.tt

```diff
@@ -1,16 +1,342 @@
+        <# } else if (arg.Type == DataType.ArrayType && arg.Type == DataType.DoubleType) { #>
+    // Create an interop array of double values.
+    unique_ptr<InteropArray> <#= arg.Name #>Array = CreateInteropArray(<#= arg.Name #>);
+        <# } else if (arg.Type == DataType.ArrayType && arg.Type == DataType.BoolType) { #>
+    // Create an interop array of bool values.
+    unique_ptr<InteropArray> <#= arg.Name #>Array = CreateInteropArray(<#= arg.Name #>);
+        <# } else if (arg.Type == DataType.PauliType) { #>
+    // Translate a PauliID value to its char representation.
+    char <#= arg.Name #>AsCharValue = TranslatePauliToChar(<#= arg.Name #>);
+        <# } else if (arg.Type == DataType.ArrayType && arg.Type == DataType.PauliType) { #>
+    // Create an interop array of Pauli values represented as chars.
+    vector<char> <#= arg.Name #>AsCharVector;
+    TranslateVector<PauliId, char>(<#= arg.Name #>, <#= arg.Name #>AsCharVector, TranslatePauliToChar);
+    unique_ptr<InteropArray> <#= arg.Name #>Array = CreateInteropArray(<#= arg.Name #>AsCharVector);
+        <# } else if (arg.Type == DataType.RangeType) { #>
+    // Create an interop range.
+    unique_ptr<InteropRange> <#= arg.Name #>Value = CreateInteropRange(<#= arg.Name #>);
+            <# } else if (arg.Type == DataType.ArrayType && arg.Type == DataType.RangeType) { #>
+    vector<InteropRange*> <#= arg.Name #>Vector;
+    TranslateVector<RangeTuple, InteropRange*>(<#= arg.Name #>, <#= arg.Name #>Vector, TranslateRangeTupleToInteropRangePointer);
+    unique_ptr<InteropArray> <#= arg.Name #>Array = CreateInteropArray(<#= arg.Name #>Vector);
+        <# } else if (arg.Type == DataType.ArrayType && arg.Type == DataType.ResultType) { #>
+    // Create an interop array of Result values.
+    unique_ptr<InteropArray> <#= arg.Name #>Array = CreateInteropArray(<#= arg.Name #>);
+        <# } else if (arg.Type == DataType.ArrayType && arg.Type == DataType.StringType) { #>
+    // Create an interop array of String values.
+    vector<const char *> <#= arg.Name #>BufferVector;
+    TranslateVector<string, const char*>(<#= arg.Name #>, <#= arg.Name #>BufferVector, TranslateStringToCharBuffer);
+    unique_ptr<InteropArray> <#= arg.Name #>Array = CreateInteropArray(<#= arg.Name #>BufferVector);
+        <# } #>
+    <# } #>
+    // Redirect the simulator output from std::cout if the --simulation-output option is present.
+    ostream* simulatorOutputStream = &cout;
+    ofstream simulationOutputFileStream;
+    if (!simulationOutputFileOpt->empty())
+    {
+        simulationOutputFileStream.open(simulationOutputFile);
+        SetOutputStream(simulationOutputFileStream);
+        simulatorOutputStream = &simulationOutputFileStream;
+    }
+
+    // Run simulation and write the output of the operation to the corresponding stream.
+    <#= entryPointOperation.Name #>(
+    <# foreach (var arg in entryPointOperation.Arguments) { #>
```

Comment:
To reduce the places where the code depends on the argument type, previous to this call, a final variable used as argument for the entry-point operation can be created with the `Interop` suffix and this logic wouldn't be dependent on argument type anymore.

Reply:
fixed

Reply:
Tracking this suggestion using this [comment](https://github.com/microsoft/qsharp-compiler/pull/929/files#r609958848).